src/odb.c

```diff
@@ -1521,6 +1521,25 @@ void git_odb_backend_data_free(git_odb_backend *backend, void *data)
 	git__free(data);
 }
 
+int git_odb_set_backend_priority(git_odb *db, size_t index, int priority)
+{
+	size_t i;
+	assert(db);
+
+	if (index > db->backends.length - 1) {
+		git_error_set(GIT_ERROR_INVALID, "invalid odb backend index");
+		return -1;
+	}
+
+	for (i = 0; i < db->backends.length; ++i){
+		if (i == index) {
+			backend_internal *internal = git_vector_get(&db->backends, i);
+			internal->priority = priority;
+		}
+	}
+	return 0;
```

Comment:
Sorting the vector after modifying priority will make this function independent. In the unit test, it is getting re-sorted since you are adding a new backend after invoking this function.

Reply:
@hackworks Great call. I've gone and updated this in a new commit to manually set the vector sorted flag to false and triggered a sort manually. Thank you 👍